src/request.c

```diff
@@ -433,6 +437,22 @@ static S3Status compose_standard_headers(const RequestParams *params,
         }                                                                   \
     } while (0)
 
+    // Host
+    if (params->bucketContext.uriStyle == S3UriStyleVirtualHost) {
+        const char *requestHostName = params->bucketContext.hostName
+                ? params->bucketContext.hostName : defaultHostNameG;
+
+        size_t len = snprintf(values->hostHeader, sizeof(values->hostHeader), "Host: %s.%s",
+                        params->bucketContext.bucketName, requestHostName);
+        if (len >= sizeof(values->hostHeader)) {
+            return S3StatusUriTooLong;
+        }
+        while (is_blank(values->hostHeader[len])) {
```

Comment:
Shouldn't this be testing is_blank(values->hostHeader[len - 1] ???

The value at [len] will start out as the  zero-terminator that snprintf put there, so this while loop will never do anything.